### PR TITLE
core: forbid conversion real->int in some cases in FileStorage

### PR DESCRIPTION
**Pull request by sovrasov**

### CODE REVIEW

File: modules/core/include/opencv2/core/cvstd.inl.hpp [hunk 1]
```diff
@@ -156,9 +156,7 @@ FileNode::operator std::string() const
 template<> inline
 void operator >> (const FileNode& n, std::string& value)
 {
-    String val;
-    read(n, val, val);
-    value = val;
+    read(n, value, std::string());
```
Comment (PkLab) [inline modules/core/include/opencv2/core/cvstd.inl.hpp]: should be managed the case `#ifndef OPENCV_NOSTL` has header

Comment (sovrasov) [inline modules/core/include/opencv2/core/cvstd.inl.hpp]: Thanks, I've forgot to check this.

File: modules/core/src/persistence.cpp [hunk 2]
```diff
@@ -7375,28 +7375,35 @@ void read(const FileNode& node, int& value, int default_value)
 {
     value = !node.node ? default_value :
     CV_NODE_IS_INT(node.node->tag) ? node.node->data.i :
-    CV_NODE_IS_REAL(node.node->tag) ? cvRound(node.node->data.f) : 0x7fffffff;
+    CV_NODE_IS_REAL(node.node->tag) ? std::numeric_limits<int>::max() : std::numeric_limits<int>::max();
```
Comment (terfendail) [inline modules/core/src/persistence.cpp]: Looks like INT_MAX will be returned in both cases. If this is intended why we even bother to check whether node is REAL?

Comment (sovrasov) [inline modules/core/src/persistence.cpp]: You're right, thanks.

**Comment by alalek** [comment]
:+1:

**Pull request merged**